src/blockdev.rs

```diff
@@ -85,8 +145,21 @@ pub struct BlkDev {
     pub swap: bool,
 }
 
-impl BlkDev {
-    pub fn get_partition_offsets(&self) -> Result<(u64, u64)> {
+impl Partition {
+    pub fn new(path: &str) -> Self {
+        Self {
+            path: path.to_string(),
+            label: None,
+            fstype: None,
+
+            parent: "".to_string(),
+            mountpoint: None,
+            swap: false,
+        }
+    }
+
+    /// Return start and end offsets within the disk.
+    pub fn get_offsets(&self) -> Result<(u64, u64)> {
```

Comment:
This is not using anything from `&self` other than `.path`. On the other hand (if I didn't miss other places), I think this is the main consumer of the `new(path: &str)` constructor above (see comment there).

I'd make this an associate method which directly takes a path (i.e. not `self`), and directly attach it to either `Disk` or `Partition`.
Alternatively, put the start+end properties as fields in `Partition` and turn this into a plain getter.